tokio/tests/sync_mpsc.rs

```diff
@@ -512,3 +512,27 @@ fn ready_close_cancel_bounded() {
 
     assert!(recv.is_woken());
 }
+
+#[tokio::test]
+async fn permit_available_not_acquired_close() {
```

Comment:
nit: it might be nice to have a comment here explaining what this test reproduces?